app/javascript/app/pages/ndc-compare-all-targets/ndc-compare-all-targets-styles.scss

```diff
@@ -19,26 +19,33 @@
     @include xy-gutters($gutter-position: ('bottom'));
   }
 
-  @media #{$tablet-portrait} {
-    @include columns((6, 6, 12));
+  @media #{$tablet-landscape} {
+    @include columns((7, 5));
+  }
+
+  margin-bottom: 20px;
+}
+
+.buttonAndSearch {
+  justify-content: flex-end;
 
-    > :first-child {
-      @include xy-gutters($gutter-position: ('bottom'));
-    }
+  @include row(12);
+
+  > * {
+    @include xy-gutters($gutter-position: ('bottom'));
   }
 
   @media #{$tablet-landscape} {
-    @include columns((7, 2, 3));
+    @include columns((6, 6));
   }
-
-  margin-bottom: 20px;
 }
 
 .compareButton {
   width: 100%;
 
-  @media #{$tablet-portrait} {
-    width: 120px;
+  @media #{$tablet-landscape} {
+    max-width: 120px;
+    min-width: 120px;
```

Comment:
I was wondering why not `width: 120px;` so I checked that, and it seems that foundation overrides stuff weirdly. I'm not a big fan of this library because I feel that we have less control over the CSS at the end, and then we need to hack our own code, but we could build using native flex or grid instead. It just the philosophical thought, no actions needed :D